### PR TITLE
gstreamer1 and friends: finally update them to current versions

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

_How about we finally update these._

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

@mascguy I borrowed what we had earlier and updated a few ports which since then had new versions.

### CODE REVIEW

File: gnome/gstreamer1-gst-plugins-bad/Portfile [hunk 1]
```diff
@@ -1,49 +1,58 @@
...
+revision            0
 description         A set of plug-ins for GStreamer that need more quality.
 long_description    GStreamer Bad Plug-ins is a set of plug-ins that aren't up to par compared \
                     to the rest. They might be close to being good quality, but they're missing \
-                    something - be it a good code review, some documentation, a set of tests, a \
-                    real live maintainer, or some actual wide use.
-license             GPL LGPL
+                    something - be it a good code review, some documentation, a set of tests, \
+                    a real live maintainer, or some actual wide use.
+license             {GPL LGPL}
```
Comment (reneeotten) [inline gnome/gstreamer1-gst-plugins-bad/Portfile:20]: there is a choice between the two licenses or do both apply?

File: gnome/gstreamer1-gst-plugins-bad/Portfile [hunk 2]
```diff
@@ -62,133 +71,87 @@ depends_lib         port:gstreamer1-gst-plugins-base \
...
+                    -Dexamples=disabled \
+                    -Dfaac=disabled \
+                    -Dfdkaac=disabled \
+                    -Dopencv=disabled \
+                    -Dsctp=disabled \
+                    -Dvulkan=disabled
 
-configure.cppflags-append   "-L${prefix}/lib"
+configure.cppflags-append   -L${prefix}/lib
 configure.cflags-append     -std=c99 -funroll-loops -fstrict-aliasing
```
Comment (reneeotten) [inline gnome/gstreamer1-gst-plugins-bad/Portfile:118]: why is this adding "c99" but then you request "2011" above?

File: gnome/gstreamer1-gst-plugins-ugly/Portfile [hunk 3]
```diff
@@ -1,67 +1,66 @@
...
+                    The license on either the plug-ins or the supporting libraries might not be \
+                    how we'd like. The code might be widely known to present patent problems.
 maintainers         nomaintainer
 categories          gnome
-platforms           darwin
 homepage            https://gstreamer.freedesktop.org/modules/${my_name}.html
 master_sites        https://gstreamer.freedesktop.org/src/${my_name}/
 distname            ${my_name}-${version}
-license             GPL-2+ LGPL-2+
+license             {GPL-2+ LGPL-2+}
```
Comment (reneeotten) [inline gnome/gstreamer1-gst-plugins-ugly/Portfile:24]: as above, a choice or do both apply?

File: net/gupnp-av/Portfile [hunk 4]
```diff
@@ -2,11 +2,11 @@
 
 PortSystem          1.0
 PortGroup           compiler_blacklist_versions 1.0
-PortGroup           gobject_introspection 1.0
+PortGroup			meson 1.0
```
Comment (reneeotten) [inline net/gupnp-av/Portfile:5]: please align

File: net/libnice/Portfile [hunk 5]
```diff
@@ -1,44 +1,50 @@
...
+PortGroup           compiler_blacklist_versions 1.0
+PortGroup           meson 1.0
 
 name                libnice
-version             0.1.17
-revision            1
-license             MPL-1.1 LGPL-2.1
+version             0.1.21
+revision            0
+license             {MPL-1.1 LGPL-2.1}
```
Comment (reneeotten) [inline net/libnice/Portfile:10]: as above

File: gnome/gstreamer1-gst-plugins-bad/Portfile (hunk 1 above)
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-bad/Portfile:20]: Thank you for pointing this out. Let me revert this. (Sorry, missed this part when porting from the earlier version.)

File: gnome/gstreamer1-gst-plugins-bad/Portfile (hunk 2 above)
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-bad/Portfile:118]: Hmm, it was already there like this. I guess we can change it to `-std=c11`.

**Comment by barracuda156** [comment]
@reneeotten I have made advised changes now.

File: gnome/gstreamer1-gst-plugins-base/Portfile [hunk 6]
```diff
@@ -123,61 +133,11 @@ platform macosx {
...
-    set merger_configure_args(i386) --build=i686-apple-${os.platform}${os.major}
-} else {
-    if {${build_arch} eq "i386"} {
-        configure.args-append \
-            --host=i686-apple-${os.platform}${os.major} \
-            --build=i686-apple-${os.platform}${os.major}
-    } elseif {${build_arch} eq "x86_64"} {
+        configure.args-replace \
+                    -Dgl=disabled \
+                    -Dgl=enabled
```
[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-base/Portfile:138]: @Gcenx Could you say which macOS versions need it? For me it builds fine without `gl-headers`.

[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-base/Portfile:138]: Ok, I will add it for those systems.

We need to wait until someone fixes `py-lxml` now which was broken by https://github.com/macports/macports-ports/commit/e825035b4dbd6acdbd2bf72f5b4024b1c649d38e (CI fail due to that)

[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-base/Portfile:138]: @Gcenx It does not install :) 
You should declare a conflict, apparently.

```
--->  Activating gl-headers @2019.1.0_0
DEBUG: Using bsdtar with HFS+ compression (if valid)
DEBUG: Using /opt/local/bin/lbzip2
DEBUG: system: /opt/local/bin/lbzip2 -d -c /opt/local/var/macports/software/gl-headers/gl-headers-2019.1.0_0.any_any.noarch.tbz2 | ( bsdtar -xvp --hfsCompression -f - )
x ./
x ./+PORTFILE
x ./+CONTENTS
x ./+DESC
x ./+STATE
x ./opt/
x ./+COMMENT
x ./opt/local/
x ./opt/local/include/
x ./opt/local/include/GL/
x ./opt/local/include/KHR/
x ./opt/local/include/KHR/khrplatform.h
x ./opt/local/include/GL/wglext.h
x ./opt/local/include/GL/glext.h
Error: Failed to activate gl-headers: Image error: /opt/local/include/GL/glext.h is being used by the active mesa port.  Please deactivate this port first, or use 'port -f activate gl-headers' to force the activation.
    while executing
"throw registry::image-error $msg"
    ("foreach" body line 47)
    invoked from within
"foreach file $imagefiles {
                set srcfile "${extracted_dir}${file}"

                # To be able to install links, we test if we can lst..."
    invoked from within
"registry::write {
            foreach file $imagefiles {
                set srcfile "${extracted_dir}${file}"

                # To be able to instal..."
DEBUG: Error code: registry::image-error
DEBUG: Backtrace: Image error: /opt/local/include/GL/glext.h is being used by the active mesa port.  Please deactivate this port first, or use 'port -f activate gl-headers' to force the activation.
    while executing
"throw registry::image-error $msg"
    ("foreach" body line 47)
    invoked from within
"foreach file $imagefiles {
                set srcfile "${extracted_dir}${file}"

                # To be able to install links, we test if we can lst..."
    invoked from within
"registry::write {
            foreach file $imagefiles {
                set srcfile "${extracted_dir}${file}"

                # To be able to instal..."
    while executing
"throw [dict get $eOptions -errorcode] [dict get $eOptions -errorinfo]"
    (procedure "_activate_contents" line 200)
    invoked from within
"_activate_contents $requested $rename_list"
    (procedure "portimage::activate" line 70)
    invoked from within
"registry_activate $subport $_inregistry_version $_inregistry_revision $_inregistry_variants $optionlist"
    (procedure "portactivate::activate_main" line 20)
    invoked from within
"$procedure $targetname"
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_gl-headers/gl-headers/main.log for details.
Error: Problem while installing gl-headers
```

[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-base/Portfile:138]: Wait, we already depend on `mesa`, which provides the headers. This cannot work. (But conflict is still needed.)

File: gnome/gstreamer1-gst-plugins-bad/Portfile [hunk 7]
```diff
@@ -62,133 +71,87 @@ depends_lib         port:gstreamer1-gst-plugins-base \
...
 }
 
-default_variants +x11
+default_variants    +x11
 
+# plugin applemedia (Apple video) now only builds on darwin 14 and later due to use
+# of constant AVQueuedSampleBufferRenderingStatusFailed introduced in OSX 10.10
 platform macosx {
-    if {![variant_isset x11] && (!${universal_possible} || ![variant_isset universal]) && ${os.major} >= 13} {
+    if {![variant_isset x11] && ${os.major} >= 13} {
```
[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-bad/Portfile]: I have changed the threshold accordingly now (to `>= 18`).

File: gnome/gstreamer1-gst-plugins-base/Portfile (hunk 6 above)
[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-base/Portfile:138]: This is nothing about `meson`: you cannot install `gl-headers` when `mesa` is active.

[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-base/Portfile:138]: Well, _we_ use x11 here :)

[the remark this answers is not in the document]
Comment (barracuda156) [reply gnome/gstreamer1-gst-plugins-base/Portfile:138]: It does, but it is accounted for: `if {![variant_isset x11]}`.

We can change defaults for new systems, but I suggest doing it separately, not in this PR. (I am fine with X11, and it is already takes forever, I do not want to be stuck with this for another year.)

Notice, I did not introduce the default of X11 here, it is already there: https://github.com/macports/macports-ports/blob/ea828893c7f358a33ed4cca992c92a4dbafae66c/gnome/gstreamer1-gst-plugins-bad/Portfile#L167

**Comment by barracuda156** [comment]
@pmetzger @reneeotten @mascguy I would like to draw attention to this. _It is the third time_ I update a dozen of ports and add needed fixes. It takes quite a bit of time on every go, both my personal and anyone who reviews this, and finally on CI.

Let us get this merged finally, please.

**Pull request merged**